lua/weapons/weapon_ace_portablemortar/shared.lua

```diff
@@ -0,0 +1,663 @@
+			if not boneid then return end
+
+			local matrix = _Owner:GetBoneMatrix(boneid)
+			if not matrix then return end
+
+			local newPos, newAng = LocalToWorld(offsetVecTube, offsetAngTube, matrix:GetTranslation(), matrix:GetAngles())
+
+			TubeModel:SetPos(newPos)
+			TubeModel:SetAngles(newAng)
+
+			newPos, newAng = LocalToWorld(offsetVecBase, offsetAngBase, matrix:GetTranslation(), matrix:GetAngles())
+
+			BaseModel:SetPos(newPos)
+			BaseModel:SetAngles(newAng)
+
+			newPos, newAng = LocalToWorld(offsetVecBipod, offsetAngBipod, matrix:GetTranslation(), matrix:GetAngles())
+
+			BipodModel:SetPos(newPos)
+			BipodModel:SetAngles(newAng)
+
+			TubeModel:SetupBones()
+			BaseModel:SetupBones()
+			BipodModel:SetupBones()
+
+		else
+			TubeModel:SetPos(self:GetPos())
+			TubeModel:SetAngles(self:GetAngles())
+			BaseModel:SetPos(self:GetPos())
+			BaseModel:SetAngles(self:GetAngles())
+			BipodModel:SetPos(self:GetPos())
+			BipodModel:SetAngles(self:GetAngles())
+		end
+
+		TubeModel:SetMaterial( "models/props_canal/metalwall005b" )
+--		TubeModel:SetColor( Color(164,190,148) )
+		TubeModel:DrawModel()
+
+		BaseModel:SetMaterial( "001wmetal" )
+--		BaseModel:SetColor( Color(0,0,0) )
+		BaseModel:DrawModel()
+
+		BipodModel:SetMaterial( "models/props_canal/canal_bridge_railing_01c" )
+		BipodModel:DrawModel()
+	end
+end
+
+
+function SWEP:SetupDataTables()
+	BaseClass.SetupDataTables(self)
+
+	self:NetworkVar("Float", 0, "FuseDelay")
+	self:NetworkVar("Float", 1, "Distance")
+	self:NetworkVar("Float", 2, "Px")
+	self:NetworkVar("Int", 3, "Py")
+	self:NetworkVar("Float", 4, "Pg")
+end
+
+function SWEP:OnPrimaryAttack()
+	self.BulletData.Owner = self:GetOwner()
+	self.BulletData.Gun = self
+
+end
+
+function SWEP:SecondaryAttack()
+
+		self:OnSecondaryAttack()
+
+		if SERVER then
+			local ZS = not self:GetZoomState()
+			self:SetZoomState(ZS)
+			self:SetOwnerZoomSpeed(ZS)
+			self:EmitSound("npc/turret_floor/click1.wav",35,75)
+		end
+end
+
+function SWEP:SetOwnerZoomSpeed(setSpeed)
+	if CLIENT then return end
+
+	local owner = self:GetOwner()
+
+	if setSpeed then
+		owner:ConCommand( "+duck" )
+		owner:SetWalkSpeed(10)
+		owner:SetRunSpeed(10) --10 is close enough to be immobile. Any lower and the crouch anim breaks because of course it does.
+
+		local nextFire = (CurTime() + self.DeployDelay)
+		self:SetNextPrimaryFire(nextFire) -- Stop reloads from resetting the fire delay
+
+	elseif self.NormalPlayerWalkSpeed and self.NormalPlayerRunSpeed then
+		owner:ConCommand( "-duck" )
+		owner:SetWalkSpeed(math.min(self.NormalPlayerWalkSpeed * self.CarrySpeedMul, self.NormalPlayerWalkSpeed))
+		owner:SetRunSpeed(math.min(self.NormalPlayerRunSpeed * self.CarrySpeedMul, self.NormalPlayerRunSpeed))
+	end
+end
+
+function SWEP:GetViewModelPosition( EyePos, EyeAng )
+	local Mul = 1
+
+	local Offset = self.IronSightsPos
+	local AngOffset = self.IronSightsAng
+
+	if ( self.IronSightsAng ) then
```

Comment:
```suggestion
	if self.IronSightsAng then
```

Unnecessary parenthesis